src/shared/utilities/timeoutUtils.ts

```diff
@@ -186,9 +224,9 @@ export async function waitUntil<T>(fn: () => Promise<T>, options: WaitUntilOptio
  *
  * @returns A Promise that returns if successful, or rejects when the Timeout was cancelled or expired.
  */
-export function waitTimeout<T, R = void, B extends boolean = true>(
+export async function waitTimeout<T, R = void, B extends boolean = true>(
```

Comment:
In what case would we define a type that extends `boolean`? This seems more confusing than just setting `allowUndefined` to true later on.

Reply:
It's because `boolean = true | false`. If you pass in `allowUndefined` as `false` then the signature reflects that by explicitly making it non-nullable. Right now I'm thinking of just cutting out that functionality and using a separate utility function `isNonNullable` since it composes far better (and makes the types easier).

This function can be retired too. Racing promises has always been pretty hacky.